### PR TITLE
feat: panel custom header

### PR DESCRIPTION
**Pull request by mfazil-eightfold**
## SUMMARY:
`panelHeader` has been introduced which accepts `React.ReactElement` as a custom header prop. 

image

## JIRA TASK (Eightfold Employees Only):
https://eightfoldai.atlassian.net/browse/ENG-26194

## CHANGE TYPE:

-   [ ] Bugfix Pull Request
-   [x] Feature Pull Request

## TEST COVERAGE:

-   [ ] Tests for this change already exist
-   [ ] I have added unittests for this change

### CODE REVIEW

File: src/components/Panel/PanelHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import { SystemUIButton } from '../Button';
+import { IconName } from '../Icon';
+
+import styles from './panel.module.scss';
+
+interface PanelHeaderProps {
+    onClose?: (e: EventType) => void;
+    title?: string;
+}
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:15]: move to panel types

File: src/components/Panel/PanelHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                    <SystemUIButton
+                        iconProps={{
+                            path: IconName.mdiClose,
+                            color: 'var(--white-color)',
+                        }}
+                        ariaLabel={'Close'}
+                        onClick={onClose}
+                        transparent
+                    />
+                )}
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:34]: add other buttons as well,

File: src/components/Panel/PanelHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import styles from './panel.module.scss';
+
+interface PanelHeaderProps {
+    onClose?: (e: EventType) => void;
+    title?: string;
+}
+
+export const PanelHeader: FC<PanelHeaderProps> = ({ onClose, title }) => {
+    const headerTitle: string = mergeClasses(['header4', styles.headerTitle]);
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:18]: 'header4' wont work, add a mixin for octuple-h4 in your css

File: src/components/Panel/Panel.types.ts [hunk 4]
```diff
@@ -159,4 +164,8 @@ export interface PanelProps extends Omit<OcBaseProps<HTMLElement>, 'title'> {
      * Custom zIndex for the panel
      */
     zIndex?: number;
+    /**
+     * Custom zIndex for the panel
+     */
+    panelHeader?: () => React.ReactElement;
```
Comment (ychhabra-eightfold) [inline src/components/Panel/Panel.types.ts:170]: no need for a funciton, if you are not hooking on button events

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

File: src/components/Panel/PanelHeader.tsx (hunk 2 above)
Comment (mfazil-eightfold) [reply src/components/Panel/PanelHeader.tsx:27]: @ychhabra-eightfold we do not have other icons at present in the library and considering the time crunch, maybe we can take this in later point? Currently we only need a close button for sms and whatsapp panel use case.

Comment (dkilgore-eightfold) [reply src/components/Panel/PanelHeader.tsx:27]: @mfazil-eightfold it only takes a moment to add the icon paths to https://github.com/EightfoldAI/octuple/blob/main/src/components/Icon/mdi.ts

Comment (dkilgore-eightfold) [reply src/components/Panel/PanelHeader.tsx:27]: The comment at the top of mdi.ts explains how to do this.

Comment (dkilgore-eightfold) [reply src/components/Panel/PanelHeader.tsx:27]: If the old component uses FA, the mapping of FA to MDI is in the figma in the comment

Comment (mfazil-eightfold) [reply src/components/Panel/PanelHeader.tsx:50]: @dkilgore-eightfold @ychhabra-eightfold i've added those changes kindly check.

File: src/components/Panel/PanelHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                    <SystemUIButton
+                        iconProps={{
+                            path: IconName.mdiClose,
+                            color: 'var(--white-color)',
+                        }}
+                        ariaLabel={'Close'}
+                        onClick={onClose}
+                        transparent
+                    />
+                )}
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:50]: {actionButtonThreeProps && (
                        <NeutralButton {...actionButtonThreeProps} />
                    )}
                    {actionButtonTwoProps && (
                        <NeutralButton {...actionButtonTwoProps} />
                    )}
                    {actionButtonOneProps && (
                        <NeutralButton {...actionButtonOneProps} />
                    )}
                    {closable && (
                        <NeutralButton
                            iconProps={{ path: closeIcon }}
                            ariaLabel={'Close'}
                            onClick={onClose}
                            {...closeButtonProps}
                        />
                    )}
                    
 can we use this pattern instead.

Comment (mfazil-eightfold) [reply src/components/Panel/PanelHeader.tsx:50]: @ychhabra-eightfold  Are you saying  to make the individual buttons to be controlled outside like, to pass custom icon etc?

File: src/components/Panel/PanelHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                {title}
+            </div>
+            <div className={styles.headerActionButtons}>
+                {!!actionButtonOneProps && (
+                    <SystemUIButton {...actionButtonOneProps} />
+                )}
+                {!!actionButtonTwoProps && (
+                    <SystemUIButton {...actionButtonTwoProps} />
+                )}
+                {!!onClose && (
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:33]:
```suggestion
                {onClose && (
```

File: src/components/Panel/PanelHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                {!!actionDefaultButtonProps && (
+                    <SystemUIButton {...actionDefaultButtonProps} />
+                )}
+                {title}
+            </div>
+            <div className={styles.headerActionButtons}>
+                {!!actionButtonOneProps && (
+                    <SystemUIButton {...actionButtonOneProps} />
+                )}
+                {!!actionButtonTwoProps && (
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:30]:
```suggestion
                {actionButtonTwoProps && (
```

File: src/components/Panel/PanelHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    return (
+        <div className={styles.logoGradientHeaderWrapper}>
+            <div className={styles.headerTitle}>
+                {!!actionDefaultButtonProps && (
+                    <SystemUIButton {...actionDefaultButtonProps} />
+                )}
+                {title}
+            </div>
+            <div className={styles.headerActionButtons}>
+                {!!actionButtonOneProps && (
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:27]:
```suggestion
                {actionButtonOneProps && (
```
I know you are used to this pattern cause of react native, but no need here

File: src/components/Panel/PanelHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    closeIcon = IconName.mdiClose,
+    title,
+    actionDefaultButtonProps,
+    actionButtonOneProps,
+    actionButtonTwoProps,
+}) => {
+    return (
+        <div className={styles.logoGradientHeaderWrapper}>
+            <div className={styles.headerTitle}>
+                {!!actionDefaultButtonProps && (
```
Comment (ychhabra-eightfold) [inline src/components/Panel/PanelHeader.tsx:21]:
```suggestion
                {actionDefaultButtonProps && (
```

File: src/components/Panel/Panel.tsx [hunk 10]
```diff
@@ -150,6 +153,11 @@ export const Panel = React.forwardRef<PanelRef, PanelProps>(
             </div>
         );
 
+        const getHeader = (): JSX.Element => {
+            if (!!PanelHeader) return PanelHeader;
+            return getDefaultHeader();
```
Comment (ychhabra-eightfold) [inline src/components/Panel/Panel.tsx:158]:
```suggestion
            return panelHeader ?? getDefaultHeader();
```

File: src/components/Panel/Panel.tsx [hunk 11]
```diff
@@ -56,6 +57,7 @@ export const Panel = React.forwardRef<PanelRef, PanelProps>(
             size = PanelSize.medium,
             visible = false,
             width,
+            PanelHeader,
```
Comment (ychhabra-eightfold) [inline src/components/Panel/Panel.tsx:60]:
```suggestion
            panelHeader,
```

**Review by ychhabra-eightfold — CHANGES_REQUESTED** [review]

**Review by ychhabra-eightfold — APPROVED** [review]

**Pull request merged**